stan/math/prim/mat/meta/broadcast_array.hpp

```diff
@@ -0,0 +1,29 @@
+#include <stan/math/prim/scal/meta/broadcast_array.hpp>
+#include <Eigen/Dense>
+#include <stdexcept>
+
+#ifndef STAN_MATH_PRIM_MAT_META_BROADCAST_ARRAY_HPP
+#define STAN_MATH_PRIM_MAT_META_BROADCAST_ARRAY_HPP
+
+namespace stan {
+  namespace math {
+    namespace internal {
+      template <typename ViewElt, typename OpElt, int R, int C>
+      class empty_broadcast_array<ViewElt, Eigen::Matrix<OpElt, R, C>> {
+      public:
+        empty_broadcast_array() {}
+        ViewElt& operator[] (int /*i*/) {
```

Comment:
Sean's probably not going to be happy I'm asking this, but I'd like this class to be formatted to match our other classes.  No space before `(` in function calls (including `operator[]`).  The open `{` is on the same line as here not on a new line as elsewhere in this pull request.   We're currently indenting namespaces like here, not flattening them like you've done elsewhere.

Reply:
OK. I fixed this.